osquery/config/additional_parsers/auto_constructed_tables.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/config.h>
+#include <osquery/database.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include "osquery/config/additional_parsers/auto_constructed_tables.h"
+#include "osquery/core/conversions.h"
+#include "osquery/sql/sqlite_util.h"
+
+namespace rj = rapidjson;
+
+namespace osquery {
+
+QueryData ATCPlugin::generate(QueryContext& context) {
+  QueryData qd;
+  std::vector<std::string> paths;
+  auto s = resolveFilePattern(path_, paths);
+  if (!s.ok()) {
+    LOG(WARNING) << "Could not glob: " << path_;
+  }
+  for (const auto& path : paths) {
+    s = genQueryDataForSqliteTable(path, sqlite_query_, qd);
+    if (!s.ok()) {
+      LOG(WARNING) << "Error Code: " << s.getCode()
+                   << " Could not generate data: " << s.getMessage();
+    }
+  }
+  return qd;
+}
+
+/// Remove these ATC tables from the registry and database
+Status ATCConfigParserPlugin::removeATCTables(
+    const std::set<std::string>& detach_tables) {
+  auto registry_table = RegistryFactory::get().registry("table");
+  for (const auto& table : detach_tables) {
+    if (registry_table->exists(table)) {
```

Comment:
A couple of questions about this block. 
* You make a check for `exists`, and exit completely if _one_ table doesn't exist, why not `LOG(WARNING)` and `continue`?
* Continuing on that last point, could you flip the logic and `continue` early sans `else` clause? This might save you a level of indentation, and could contribute to the next point
* What if a table exists, but the `getDatabaseValue` fails? Perhaps you could bring the `getDatabaseValue` out of the `if`, and should it fail emit a `LOG(WARNING)` that the DB lookup failed?

Reply:
1) Good idea
2) We don't want to flip it because we want the side effect of removing from the db.
3) If the table exists but get fails then the table is a core table (registered but not in the db) so that's the message we create